orekit/src/maneuvers.py

```diff
@@ -0,0 +1,68 @@
+""" Simulate Orekit maneuvers as poliastro ones """
+
+import numpy as np
+from astropy import units as u
+from numpy.linalg import norm
+
+from utils import vm  # noqa: F401 # isort: split
+from org.hipparchus.geometry.euclidean.threed import Vector3D
+from org.orekit.attitudes import LofOffset
+from org.orekit.forces.maneuvers import ImpulseManeuver
+from org.orekit.frames import LOFType
+from org.orekit.orbits import OrbitType, PositionAngle
+from org.orekit.propagation.events import PositionAngleDetector
+
+
+class OrekitManeuver:
+    """ Class for modeling impulsive maneuvers """
+
+    def __init__(self, burns, attitude_provider):
+        """ Initializes the maneuver """
+
+        # Save the ImpulsiveManeuver instances and the attitude
+        self.burns = burns
+        self.attitude_provider = attitude_provider
+
+        # Allocate the final dV vectors
+        self.impulses = [dV.getDeltaVSat().toArray() * u.m / u.s for dV in self.burns]
+
+    @classmethod
+    def hohmann(cls, orekit_ss, rf_norm, ISP=float(300)):
+        """ Computes required impulses for a Hohmann transfer """
+
+        # Get required parameters and auxiliary ones
+        k = orekit_ss.attractor.k
+        r0_vec, v0_vec = orekit_ss.rv()
+
+        # Initial position and velocity magnitude
+        r0_norm, v0_norm = [norm(vec) for vec in (r0_vec, v0_vec)]
+        a_trans = (r0_norm + rf_norm) / 2
+
+        # Compute the magnitude of Hohmann's deltaV
+        dv_a = np.sqrt(2 * k / r0_norm - k / a_trans) - v0_norm
+        dv_b = np.sqrt(k / rf_norm) - np.sqrt(2 * k / rf_norm - k / a_trans)
+
+        # Local orbit frame: X-axis aligned with velocity, Z-axis towards momentum
+        attitude_provider = LofOffset(orekit_ss._frame, LOFType.TNW)
+
+        # Convert previous magnitudes into vectors within the TNW frame, which
+        # is aligned with local velocity vector
+        dVa_vec, dVb_vec = [
+            Vector3D(float(dV.to(u.m / u.s).value), float(0), float(0))
+            for dV in (dv_a, dv_b)
+        ]
+
+        # Define the events when that trigger the impulses
+        at_periapsis = PositionAngleDetector(
+            OrbitType.KEPLERIAN, PositionAngle.TRUE, float(0)
+        )
+        at_apoapsis = PositionAngleDetector(
+            OrbitType.KEPLERIAN, PositionAngle.TRUE, float(np.pi)
+        )
+
```

Comment:
The `PositionAngleDetector` depends on the kind of orbit. After applying a Hohmann maneuver, final orbit becomes circular and this can introduce non-expected values in the final result. Therefore, it is better to use the `ApsideDetector` trigger.